File: 3442-maximum-difference-between-even-and-odd-frequency-i/3442-maximum-difference-between-even-and-odd-frequency-i.java
import java.util.*;

public class Solution {
    public int maxDifference(String s) {
        int[] freq = new int[26];
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        int odd = 0;
        int even = s.length();
        for (int n : freq) {
            if ((n & 1) == 1) {
                odd = Math.max(odd, n);
            } else if (n != 0) {
                even = Math.min(even, n);
            }
        }

        return odd - even;
    }
}

File: 3442-maximum-difference-between-even-and-odd-frequency-i/3442-maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        from collections import Counter
        
        freq = Counter(s)
        odd = 0
        even = len(s)

        for count in freq.values():
            if count % 2 == 1:
                odd = max(odd, count)
            elif count != 0:
                even = min(even, count)

        return odd - even